File: sample-chat-obj-c/sample-chat/InfoScreen/InfoScreenViewController/InfoTableViewController.m
//
//  InfoTableViewController.m
//  sample-chat
//
//  Created by Injoit on 2/25/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

#import "InfoTableViewController.h"
#import "InfoModel.h"
#import "InfoTableViewCell.h"
#import <Quickblox/Quickblox.h>

NSString *const kInfoTableViewCellId = @"InfoTableViewCell";
NSString *const kLogoTableViewCellId = @"QBLogoTableViewCell";


@interface InfoTableViewController ()
@property (strong, nonatomic) NSMutableArray<InfoModel*>* infoModels;
@end

@implementation InfoTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIBarButtonItem *backButtonItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"chevron"]
                                                                       style:UIBarButtonItemStylePlain
                                                                      target:self
                                                                      action:@selector(didTapBack:)];
    
    self.navigationItem.leftBarButtonItem = backButtonItem;
    backButtonItem.tintColor = UIColor.whiteColor;
    
    [self setupTableView];
}

- (void)didTapBack:(UIButton *)sender {
    [self.navigationController popViewControllerAnimated:YES];
}

- (void)setupTableView {
    self.infoModels = [NSMutableArray new];
    
    NSString *appName = NSBundle.mainBundle.infoDictionary[@"CFBundleName"];
    InfoModel *appNameModel = [[InfoModel alloc] init];
    appNameModel.title = @"Application version";
    appNameModel.info = [NSString stringWithFormat:@"%@", appName];
    [self.infoModels addObject:appNameModel];
    
    NSString *appVersion = NSBundle.mainBundle.infoDictionary[@"CFBundleShortVersionString"];
    InfoModel *appVersionModel = [[InfoModel alloc] init];
    appVersionModel.title = @"Application version";
    appVersionModel.info = [NSString stringWithFormat:@"%@", appVersion];
    [self.infoModels addObject:appVersionModel];
    
    InfoModel *quickBloxSdkVersionModel = [[InfoModel alloc] init];
    quickBloxSdkVersionModel.title = @"QuickBlox SDK version";
    quickBloxSdkVersionModel.info = [NSString stringWithFormat:@"%@", QuickbloxFrameworkVersion];
    [self.infoModels addObject:quickBloxSdkVersionModel];
    
    InfoModel *appIDModel = [[InfoModel alloc] init];
    appIDModel.title = @"Application ID";
    appIDModel.info = [NSString stringWithFormat:@"%@", @(QBSettings.applicationID)];
    [self.infoModels addObject:appIDModel];
    
    InfoModel *authKeyModel = [[InfoModel alloc] init];
    authKeyModel.title = @"Auhtorization key";
    authKeyModel.info = [NSString stringWithFormat:@"%@", QBSettings.authKey];
    [self.infoModels addObject:authKeyModel];
    
    InfoModel *authSecretModel = [[InfoModel alloc] init];
    authSecretModel.title = @"Auhtorization secret";
    authSecretModel.info = [NSString stringWithFormat:@"%@", QBSettings.authSecret];
    [self.infoModels addObject:authSecretModel];
    
    InfoModel *accountKeyModel = [[InfoModel alloc] init];
    accountKeyModel.title = @"Account key";
    accountKeyModel.info = [NSString stringWithFormat:@"%@", QBSettings.accountKey];
    [self.infoModels addObject:accountKeyModel];
    
    InfoModel *apiDomainModel = [[InfoModel alloc] init];
    apiDomainModel.title = @"API domain";
    apiDomainModel.info = [NSString stringWithFormat:@"%@", QBSettings.apiEndpoint];
    [self.infoModels addObject:apiDomainModel];
    
    InfoModel *chatDomainModel = [[InfoModel alloc] init];
    chatDomainModel.title = @"Chat domain";
    chatDomainModel.info = [NSString stringWithFormat:@"%@", QBSettings.chatEndpoint];
    [self.infoModels addObject:chatDomainModel];
    
    NSString *qaVersion = NSBundle.mainBundle.infoDictionary[@"CFBundleVersion"];
    InfoModel *qaVersionModel = [[InfoModel alloc] init];
    qaVersionModel.title = @"QA version";
    qaVersionModel.info = [NSString stringWithFormat:@"%@", qaVersion];
    [self.infoModels addObject:qaVersionModel];
    
    InfoModel *logoModel = [[InfoModel alloc] init];
    logoModel.title = @"logo";
    logoModel.info = @"logo";
    [self.infoModels addObject:logoModel];
    
    [self.tableView reloadData];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.infoModels.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    BOOL isLast = indexPath.row == self.infoModels.count - 1;
    if (isLast) {
        UITableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:kLogoTableViewCellId];
        return cell;
    }
    InfoTableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:kInfoTableViewCellId];
    InfoModel *model = self.infoModels[indexPath.row];
    [cell applyInfo:model];
    return cell;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    BOOL isLast = indexPath.row == self.infoModels.count - 1;
    return isLast ? 80.0f : 54.0f;
}

@end

File: sample-push-notifications-swift/sample-push-notifications-swift/InfoScreenViewController/InfoTableViewController.swift
//
//  InfoTableViewController.swift
//  sample-push-notifications-swift
//
//  Created by Injoit on 12/30/18.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit
import Quickblox

struct InfoCellConstant {
    static let infoTableViewCellId = "InfoTableViewCell"
    static let logoTableViewCellId = "QBLogoTableViewCell"
}

class InfoTableViewController: UITableViewController {
    
    //MARK: - Properties
    var infoModels = [InfoModel]()
    
    //MARK: - Life Cycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let backButtonItem = UIBarButtonItem(image: UIImage(named: "chevron"),
                                                           style: .plain,
                                                           target: self,
                                                           action: #selector(didTapBack(_:)))
        navigationItem.leftBarButtonItem = backButtonItem
        backButtonItem.tintColor = .white
        
        setupTableView()
    }
    
    @objc func didTapBack(_ sender: UIBarButtonItem) {
        navigationController?.popViewController(animated: true)
    }
    
    //MARK: - Internal Methods
    private func setupTableView() {
        let appName = Bundle.main.infoDictionary?["CFBundleName"] as? String
        let appNameModel = InfoModel()
        appNameModel.title = "Application name"
        appNameModel.info = appName ?? ""
        infoModels.append(appNameModel)
        
        let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String
        let appVersionModel = InfoModel()
        appVersionModel.title = "Application version:"
        appVersionModel.info = "\(appVersion ?? "")"
        infoModels.append(appVersionModel)
        
        let quickBloxSdkVersionModel = InfoModel()
        quickBloxSdkVersionModel.title = "QuickBlox SDK version:"
        quickBloxSdkVersionModel.info = "\(QuickbloxFrameworkVersion)"
        infoModels.append(quickBloxSdkVersionModel)
        
        let appIDModel = InfoModel()
        appIDModel.title = "Application ID:"
        appIDModel.info = "\(QBSettings.applicationID)"
        infoModels.append(appIDModel)
        
        let authKeyModel = InfoModel()
        authKeyModel.title = "Auhtorization key:"
        authKeyModel.info = "\(QBSettings.authKey ?? "")"
        infoModels.append(authKeyModel)
        
        let authSecretModel = InfoModel()
        authSecretModel.title = "Auhtorization secret:"
        authSecretModel.info = "\(QBSettings.authSecret ?? "")"
        infoModels.append(authSecretModel)
        
        let accountKeyModel = InfoModel()
        accountKeyModel.title = "Account key:"
        accountKeyModel.info = "\(QBSettings.accountKey ?? "")"
        infoModels.append(accountKeyModel)
        
        let apiDomainModel = InfoModel()
        apiDomainModel.title = "API domain:"
        apiDomainModel.info = "\(QBSettings.apiEndpoint ?? "")"
        infoModels.append(apiDomainModel)
        
        let chatDomainModel = InfoModel()
        chatDomainModel.title = "Chat domain:"
        chatDomainModel.info = "\(QBSettings.chatEndpoint ?? "")"
        infoModels.append(chatDomainModel)
        
        let qaVersion = Bundle.main.infoDictionary?["CFBundleVersion"] as? String
        let qaVersionModel = InfoModel()
        qaVersionModel.title = "QA version"
        qaVersionModel.info = qaVersion ?? ""
        infoModels.append(qaVersionModel)
        
        let logoModel = InfoModel()
        logoModel.title = "logo"
        logoModel.info = "logo"
        infoModels.append(logoModel)
        
        tableView.reloadData()
    }
    
    // MARK: - Table view data source
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return infoModels.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let isLast = indexPath.row == infoModels.count - 1
        if isLast == true {
            let cell = tableView.dequeueReusableCell(withIdentifier: InfoCellConstant.logoTableViewCellId, for: indexPath)
            return cell
        }
        guard let cell = tableView.dequeueReusableCell(withIdentifier: InfoCellConstant.infoTableViewCellId)
            as? InfoTableViewCell else {
                return UITableViewCell()
        }
        let model = infoModels[indexPath.row]
        cell.applyInfo(model: model)
        return cell
    }
    
    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        let isLast = indexPath.row == infoModels.count - 1
        return isLast ? 80.0 : 54.0
    }
}
